rootfs/etc/nginx/lua/crypto/cipher.lua

```diff
@@ -0,0 +1,44 @@
+local sha1_crypto = require("crypto.sha1")
+local md5_crypto = require("crypto.md5")
+local ffi = require "ffi"
+local ffi_new = ffi.new
+local ffi_str = ffi.string
+local C = ffi.C
+
+local sha1 = sha1_crypto:new()
+local md5 = md5_crypto:new()
+
+local _M = {}
+
+function _M.md5_digest(raw, eof)
+  md5:update(raw)
+  if eof then
+    return md5:final()
+  end
+  return nil
+end
+
+function _M.sha1_digest(raw, eof)
+  sha1:update(raw)
+  if eof then
+      return sha1:final()
+  end
+  return nil
+end
+
+-- Copyright (C) by Yichun Zhang (agentzh)
```

Comment:
why is this in the middle?

Reply:
This is the original file: https://github.com/Shopify/nginx-routing-modules/blob/1962ec1fc78b2621ea0f122031819a9d54ec9882/lua/vendor/resty/string.lua

I'll separate my code and this and then move to vendor folder.

Reply:
> I'll separate my code and this and then move to vendor folder.

👍